### PR TITLE
[Marketplace] Order Filtering By Status

### LINKED ISSUE #2357: Filter Orders By Status
Users with permissions to view orders dashboard should be able to quickly filter the list of orders by the order status.

From a dropdown, users should be able to filter orders by status. Ideal scenario permits users to filter by multiple statuses simultaneously. (e.g. select orders that are "New" and "Processing" and see only orders with those statuses)

If users `search` while a filter is active, the search should take the filter into account.

Users should be able to clear all filters easily as well (#????)

##### New 
-   Web
-   POS (soon)
-   Phone (soon)

##### Processing
-   Ready 
-   Picking 
-   Picked 
-   Packing 
-   Packed 
-   Labeled (Ready For Shipping / Pickup)

##### Shipped 
-   In Transit 
    -   Last Location
-   Exception
    -   Delay (Weather, Error)
    -   Lost
    -   Damaged
    -   Redirected

##### Delivered / Completed
-   Delivered [Location (front door, porch, etc)

##### Returning 
-   Return Requested
-   Refunded, awaiting return
-   Return Shipped
-   Return Exception
-   Return Delivered

##### Cancelled
-   Cancelled By Customer
-   Cancelled By Vendor
-   Partially Cancelled

Extends #2152 with specifications.

Annotated Designs in Zeplin:
Web: https://zpl.io/1r0Vvm
App: zpl://screen?pid=58af8900737c137280322bd6&sid=58fa9b08c1ee69df600c0469

Related to: #2046

### PR DESCRIPTION
**Pull request by joykare**
Resolves #2357 

*Description*
Users with permissions to view orders dashboard should be able to quickly filter the list of orders by the order status.

*How to test*
- Clone branch
- Create multiple orders
- Have some orders in different stages of the order workflow ["New", "Approved", "Captured", "Completed", "Cancelled"]
- Use filters observe orders display according to filter
- Click on left-most icon to clear filter [Observe all your orders displayed]

### CODE REVIEW

**Comment by joykare** [comment]
@rymorgan one more thing. Users ideally should be able to filter by multiple statuses simultaneously. So I think we need to have a way to show that the individual statuses have been selected/deselected.

**Comment by rymorgan** [comment]
filter-feedback

For the date range filter can you use this design? (it's yet to be implemented anywhere) 
https://zpl.io/a8K3pdV

File: imports/plugins/core/orders/client/templates/orders.js [hunk 1]
```diff
@@ -1,10 +1,30 @@
 import { Template } from "meteor/templating";
-import OrdersContainer from "../containers/ordersContainer";
+// import { Reaction } from "/client/api";
```
Comment (zenweasel) [inline imports/plugins/core/orders/client/templates/orders.js]: Can we remove this commented code?

File: package.json [hunk 2]
```diff
@@ -26,6 +26,7 @@
     "autoprefixer": "^7.1.2",
     "autosize": "^4.0.0",
     "babel-runtime": "^6.23.0",
+    "bcrypt": "^1.0.2",
```
Comment (zenweasel) [inline package.json]: Nope. No bcrypt.

**Comment by joykare** [comment]
@zenweasel good for another check

**Comment by zenweasel** [comment]
If I filter for Completed, I get both Cancelled and Completed
cart_completed

**Comment by zenweasel** [comment]
Except for the Completed/Cancelled item noted above, seems to work well.

**Comment by joykare** [comment]
A cancelled order is completed that's the reasoning behind that. But if having cancelled orders not part of the completed ones, I could do that.

**Comment by kieha** [comment]
I think `Canceled` orders should not be included in the `Completed` ones while filtering for completed. It might be a bit confusing.

**Comment by rymorgan** [comment]
Yep, canceled orders should not be included in completed filter :)

**Comment by rymorgan** [comment]
@joykare I thought we fixed all the responsive issues? It's still basically not working responsively. 
cart_completed_and_inbox_ _morgan_ry_gmail_com_and_reaction_ _node_ _node__usr_local_bin_rc_ _152x48

**Comment by joykare** [comment]
Cancelled/Completed orders issue fixed @kieha  @zenweasel 

@rymorgan being fixed on this issue #2617

**Comment by joykare** [comment]
@zenweasel

**Pull request merged**